OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -120,6 +120,12 @@ public override bool Tick(Actor self)
 			// and resume the activity after reloading if AbortOnResupply is set to 'false'
 			if (rearmable != null && !useLastVisibleTarget && attackAircraft.Armaments.All(x => x.IsTraitPaused || !x.Weapon.IsValidAgainst(target, self.World, self)))
 			{
+				// HACK: The gameplay expectation for AbortOnResupply is to cancel all queued activities, but only
+				// if the attack was explicitly ordered (i.e. FlyAttack is a top-level activity).
+				// Implicit orders (i.e. FlyAttack is a child activity) should not be disrupted
+				if (attackAircraft.Info.AbortOnResupply && NextActivity != null && self.CurrentActivity == this)
```

Comment:
The `self.CurrentActivity` check doesn't work if an attack order is queued after another activity while having no ammo. CurrentActivity isn't updated until the end of `RunActivity`, so this check will always be false for the first tick.

Reply:
The only time `FlyAttack` is called as a child activity, is when it is called by `AttackMoveActivity`. So it would be cleaner IMO to explicitly define this behaviour there, by passing a boolean argument down to the childactivity. This avoids the issue with checking `self.CurrentActivity == this` entirely, which is a hack in itself.

Reply:
I considered that, but the problem is that `AttackMoveActivity` works through an interface, and exposing this workaround to all attack traits is IMO a bigger hack than the (now updated) check here.

If this still crosses our collective hack tolerance threshold I would rather abandon the idea completely.